### PR TITLE
Tweak shuttle movement

### LINKED ISSUE #5465: Thruster balancing
Ideally:
- Work out a decent way to do the speedcap
- Buff impulses (needs speedcap to avoid a 3 tile shuttle from being unplayably fast)
- Potentially make the impulses logarithmic up to a certain point or something

### LINKED ISSUE #6248: Need debug thrusters
That don't require open tile / power and aren't constructible.

### PR DESCRIPTION
**Pull request by Radrark**
## About the PR 

Adds a linear and an angular speed cap to shuttles. It does so by augmenting the damping and then, by the power of math (and the physics engine), the velocity naturally reaches a maximum. This maximum is the so-called terminal velocity and it depends on the damping coefficient and the acceleration of the shuttle according to this formula: **terminal velocity = acceleration / damping**.
This means that every shuttle will have a different terminal velocity depending on the amount of thrust they exert. In order to avoid very high terminal velocities we hard cap the acceleration.

The result is imo nice movement for both small and big shuttles (except maybe rotation for big shuttles).
Still, the constants may be tweaked if needed.

closes #5465 
closes #6248 
maybe #6249

**Changelog**

:cl:
- tweak: Improved the movement of shuttles.

### CODE REVIEW

File: Content.Server/Shuttles/EntitySystems/ThrusterSystem.cs [hunk 1]
```diff
@@ -118,7 +118,8 @@ private void OnTileChange(object? sender, TileChangedEventArgs e)
 
                     foreach (var ent in _mapManager.GetGrid(e.NewTile.GridIndex).GetAnchoredEntities(checkPos))
                     {
-                        if (!EntityManager.TryGetComponent(ent, out ThrusterComponent? thruster) || thruster.Type == ThrusterType.Angular) continue;
+                        if (!EntityManager.TryGetComponent(ent, out ThrusterComponent? thruster) || thruster.Type == ThrusterType.Angular ||
+                            !thruster.RequireSpace) continue;
```
Comment (metalgearsloth) [inline Content.Server/Shuttles/EntitySystems/ThrusterSystem.cs:122]: Realistically these angular checks can be replaced and just use with RequireSpace.

File: Content.Shared/CCVar/CCVars.cs [hunk 2]
```diff
@@ -557,6 +557,24 @@ public sealed class CCVars : CVars
         public static readonly CVarDef<float> ShuttleDockSpeedCap =
             CVarDef.Create("shuttle.dock_speed_cap", 5f, CVar.SERVERONLY);
 
+        public static readonly CVarDef<float> ShuttleMaxLinearAcc =
+            CVarDef.Create("shuttle.max_linear_acc", 13f, CVar.SERVERONLY);
+
+        public static readonly CVarDef<float> ShuttleMaxAngularAcc =
+            CVarDef.Create("shuttle.max_angular_acc", 3f, CVar.SERVERONLY);
```
Comment (metalgearsloth) [inline Content.Shared/CCVar/CCVars.cs:564]: Call it Impulse instead of Acc

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
I don't think https://github.com/space-wizards/space-station-14/issues/6249 is resolved as there's no minimum threshold yet to prevent something like saltern from moving.

I still need to playtest ingame if anyone else sees this PR.

File: Content.Shared/CCVar/CCVars.cs (hunk 2 above)
Comment (Radrark) [reply Content.Shared/CCVar/CCVars.cs:564]: I mean it's not really a cap on impulse, because impulse depends on mass.

**Comment by ike709** [comment]
You need `Closes` before each issue.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
- Thrusters are very overtuned in this PR, it takes less than a second to hit the absolute speed cap so you only need 1 thruster for medium sized ships. Saltern moves from a single thruster and hits speedcap in a couple of seconds with 2.
- Angular speedcap needs to be higher as it also gets hit very easily. If this is a problem then the angular velocity needs capping and the inertia needs increasing because Saltern hits it with a couple gyroscopes. I'm fine with it rotating slowly to reflect the inertia but it shouldn't be able to be hit so easily.

**Comment by Radrark** [comment]
I nerfed the thrusters so big ships need more thrusters to move faster. About the shuttles hitting max angular speed quickly I think it's fine like that because it makes rotating easier. All in all, the point of the PR is to make controlling shuttles easier so hitting max speed quickly shouldn't be a problem if piloting feels good.

**Comment by metalgearsloth** [comment]
I'm still not a fan of thrusters primarily being used to increase speedcap as it feels very sluggish to me as there's no momentum, especially with rotation, and doesn't feel like a physics sim, I'm also not sure how it would fit nicely with shuttle collisions when it's so easy to repeatedly hit something when acceleration is constant with mass.

**Comment by ElectroJr** [comment]
> All in all, the point of the PR is to make controlling shuttles easier so hitting max speed quickly shouldn't be a problem if piloting feels good.

I'm open to making small-medium shuttles more manoeuvrable / responsive, But IMO stations should definitely have a sense of momentum, and this seems to just get rid of that, they can go from to to 100(%) and back again very quickly.

But I also have no idea what people really want / what is fun for whole-station piloting. It might be the case that fun ship v ship game modes would just require manoeuvrable stations.

**Comment by moonheart08** [comment]
Stations and large shuttles definitely should feel like flying a building. If you want to give shuttles an advantage, maybe add inertial dampers? they're a p common scifi trope.

**Comment by metalgearsloth** [comment]
Probably gonna need more bikeshedding but I would like the rest of the changes on master (debug thruster, requiresspace, and cvars). I'm fine with cherrypicking it from the PR if you're okay with it @Radrark

**Comment by Radrark** [comment]
Okay, I changed it now all shuttles have the same max speed and adding thrusters makes them reach it faster. Also made the damping an actual force instead of an acceleration.

With these changes I gave big ships some momentum and at the same time I tried to conserve the maneuverability of the smaller shuttles.

**Comment by Radrark** [comment]
I nerfed rotation, it's still a little bit weird though. I think that having gyroscopes with different thrust values would be better in order to balance rotation for both big and small ships. I mean having low grade gyroscopes for small ships and high grade for the big ones. But I guess that's for another PR.

**Comment by Radrark** [comment]
I changed it so now it caps angular momentum in order to avoid high speeds at the edges of big ships however this makes them really hard to rotate but I think it's fine.

File: Resources/Prototypes/Entities/Structures/Shuttles/thrusters.yml [hunk 3]
```diff
@@ -69,7 +94,8 @@
   components:
   - type: Thruster
     thrusterType: Angular
-    impulse: 200
+    requireSpace: false
+    thrust: 23000
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Structures/Shuttles/thrusters.yml:98]:
```suggestion
    thrust: 5000
```

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**